### ISSUE TITLE
Create a States system

### ISSUE THREAD

**Issue by Sygmei** [Sygmei/ObEngine#172]
Once the new Scene system is done, each Scene could be a state

**Comment by TheMaverickProgrammer** [comment]
I recommend [Swoosh](https://github.com/TheMaverickProgrammer/Swoosh), It's C++ and built on top of SFML. It also handles screen transition effects

**Comment by Sygmei** [comment]
I remember checking your library, I'll investigate it a bit :) Thanks for the suggestion, is it fully integrated to SFML or is it possible to use it without SFML ?

**Comment by TheMaverickProgrammer** [comment]
It's currently built with SFML and zero other dependencies. From what I understand, ObEngine is built with SFML too so there will be no library issues.

**Comment by Sygmei** [comment]
Yup, ÖbEngine also uses SFML but I will maybe change that later, that's why I try to avoid dependencies of dependencies (TGUI is a good example, I'll probably switch to Qt).

**Comment by TheMaverickProgrammer** [comment]
Ok. I've been asked already to port Swoosh. Maybe it'll be in everyone's interest if I rewrite it for the next media layer so ObEngine can use it. Were you planning SDL?

**Comment by Sygmei** [comment]
https://github.com/TheMaverickProgrammer/Swoosh has a nice composer system
We might be able to port it partially minus the transition system (which is the only part that requires SFML IIRC).
The result could be a `SceneComposer` which loads, unloads and updates Scenes
